add GetEvent to get unmarshaled data from events

jpeeler opened the pull request:
This adds a GetEvent method so that users can work with device types without worrying about JSON conversions and parsing the devices path.

The Event type has been modified to store bytes instead of strings, since that is what's required for unmarshaling. To help avoid exposing this change to the user a String method has also been added.

Simple example usage with exactly one device, ignoring errors:

```go
 eventType, _, data, _ := event.GetEvent(false)
 if eventType == nest.KeepAlive {
   return
 }
 mynest, _ := data.(*device.Thermostat)
 fmt.Printf("Ambient temperature:\n", mynest.AmbientTemperatureF)
```

I've really only tested this with my Nest thermostat, so it's possible I've messed something up (well, really that's always possible)!

Commit:
add GetEvent to get unmarshaled data from events

This adds a GetEvent method so that users can work with device types
without worrying about JSON conversions and parsing the devices path.

The Event type has been modified to store bytes instead of strings,
since that is what's required for unmarshaling. To help avoid exposing
this change to the user a String method has also been added.

Simple example usage with exactly one device, ignoring errors:

 eventType, _, data, _ := event.GetEvent(false)
 if eventType == nest.KeepAlive {
   return
 }
 mynest, _ := data.(*device.Thermostat)
 fmt.Printf("Ambient temperature:\n", mynest.AmbientTemperatureF)

jtsiros commented:
Thanks for the contribution! There is a convenience method for each service for device-specific streaming called `Stream()`. This is really helpful, but I'm wondering if it would be best to separate out the filtering into each device service `ThermostatService`, `SmokeCoAlarmService`, `CameraService` just to keep `event.go` simple. Maybe some kind of decorator pattern on `Stream()` that filters out the specific device event types?

bitshiftxr committed:
Update event.go

Change ERROR to error as per REST Streaming example at https://developers.nest.com/guides/api/rest-streaming-guide#subscribe_to_changes_in_structures_and_devices to keep types consistent.

bitshiftxr committed:
Add Unit Tests

Adds unit tests for getEvents.

bitshiftxr committed:
Additional Code Cleanup

Removes dead err check on json.Unmarshal after `path` has been parsed. Any potential err unmarshal should be caught on L72. Also removes printing as part of the method contract.

bitshiftxr committed:
Add Additional Test Cases

Adds additional test cases to cover reading raw events and GetEvents.
